src/portable/portable.c

```diff
@@ -1187,6 +1200,19 @@ int portable_attach(
                         return r;
         }
 
+        if (hashmap_isempty(unit_files)) {
+                _cleanup_free_ char *extensions = strv_join(extension_image_paths, ", ");
+                if (!extensions)
+                        return -ENOMEM;
+
+                return sd_bus_error_setf(error,
+                                         SD_BUS_ERROR_INVALID_ARGS,
+                                         "Couldn't find any matching unit files in image '%s%s%s', refusing.",
+                                         image->path,
+                                         isempty(extensions) ? "" : "' or any of its extensions '",
+                                         isempty(extensions) ? "" : extensions);
+        }
```

Comment:
it feels like a good chunk of this code could be split out into a helper function shared by portable_extract() and portable_attach() (maybe called portable_prepare() or so?)